homeassistant/helpers/device_registry.py

```diff
@@ -43,29 +44,31 @@ def async_get_device(self, identifiers: str, connections: tuple):
         """Check if device is registered."""
         for device in self.devices:
             if any(iden in device.identifiers for iden in identifiers) or \
-                    any(conn in device.connection for conn in connections):
+                    any(conn in device.connections for conn in connections):
                 return device
         return None
 
     @callback
     def async_get_or_create(self, identifiers, manufacturer, model,
-                            connection, *, name=None, sw_version=None):
+                            connections, *, name=None, sw_version=None,
+                            config_entry=None):
         """Get device. Create if it doesn't exist."""
-        device = self.async_get_device(identifiers, connection)
+        device = self.async_get_device(identifiers, connections)
+
+        if device is None:
+            device = DeviceEntry(
+                connections=connections,
+                identifiers=identifiers,
+                manufacturer=manufacturer,
+                model=model,
+                name=name,
+                sw_version=sw_version,
+            )
+            self.devices.append(device)
+
+        if config_entry is not None:
+            device.config_entries.add(config_entry)
 
-        if device is not None:
-            return device
-
-        device = DeviceEntry(
-            identifiers=identifiers,
-            manufacturer=manufacturer,
-            model=model,
-            connection=connection,
-            name=name,
-            sw_version=sw_version
-        )
-
-        self.devices.append(device)
         self.async_schedule_save()
```

Comment:
We should only save if the device has changed.

Reply:
Allright